flow/flow.go

```diff
@@ -499,6 +545,167 @@ func (f *Flow) updateMetricsWithNetworkLayer(packet *gopacket.Packet) error {
+		var synData []byte
+		if app := (*packet).ApplicationLayer(); app != nil {
+			synData = app.Payload()
+		} else {
+			logging.GetLogger().Notice("Capture SYN failed to parse app data. leaving empty")
+		}
+		if f.Network.A == srcIP {
+			if f.TCPFlowMetric.ABSynStart == 0 {
+				f.TCPFlowMetric.ABSynStart = captureTime
+				f.TCPFlowMetric.ABSynTTL = timeToLive
+				f.TCPFlowMetric.ABSynData = synData
+			} else {
+				logging.GetLogger().Notice("Duplicate SYNCapture AB", f.TCPFlowMetric.ABSynData)
+			}
+		} else {
+			if f.TCPFlowMetric.BASynStart == 0 {
+				f.TCPFlowMetric.BASynStart = captureTime
+				f.TCPFlowMetric.BASynTTL = timeToLive
+				f.TCPFlowMetric.BASynData = synData
+			} else {
+				logging.GetLogger().Notice("Duplicate SYNCapture BA", f.TCPFlowMetric.BASynData)
+			}
+		}
+	case tcpPacket.FIN:
+		if f.Network.A == srcIP {
+			f.TCPFlowMetric.ABFinStart = captureTime
+		} else {
+			f.TCPFlowMetric.BAFinStart = captureTime
+		}
+	case tcpPacket.RST:
+		if f.Network.A == srcIP {
+			f.TCPFlowMetric.ABRstStart = captureTime
+		} else {
+			f.TCPFlowMetric.BARstStart = captureTime
+		}
+	}
+
+	return nil
+}
+
+func (f *Flow) updateMetricsBySeq(packet *gopacket.Packet) error {
+	transportLayer := (*packet).Layer(layers.LayerTypeTCP)
+	transportPacket, transp_ok := transportLayer.(*layers.TCP)
+        if !transp_ok {
+		return nil
+	}
+
+        var expectedSeq uint32
+
+	ipv4Layer := (*packet).Layer(layers.LayerTypeIPv4)
+	ipv4Packet, ok := ipv4Layer.(*layers.IPv4)
+	if ok {
+		if f.Network.A == ipv4Packet.SrcIP.String() {
+			if transportPacket.SYN {
+				f.state.aBExpectedSeq = getExpectedSeq(ipv4Packet, transportPacket)
+				logging.GetLogger().Notice("SYN AB", f.state.aBExpectedSeq)
+				return nil
+			}
+			expectedSeq = f.state.aBExpectedSeq
+		} else {
+			if transportPacket.SYN {
+				f.state.bAExpectedSeq = getExpectedSeq(ipv4Packet, transportPacket)
+				logging.GetLogger().Notice("SYN BA", f.state.bAExpectedSeq)
+				return nil
+			}
+			expectedSeq = f.state.bAExpectedSeq
+		}
+	} else {
+		logging.GetLogger().Error("Error retrieving IP layer")
+	}
+
+	length := uint32Distance(transportPacket.Seq, expectedSeq)
+
+	if transportPacket.Seq < TCPMaxSegmentSize {
+		logging.GetLogger().Notice("Overlap", transportPacket.Seq, expectedSeq)
+	}
+
+	if length > 0x80010000 {
+		logging.GetLogger().Notice("Detected retransmission", transportPacket.Seq)
+		return nil
+	}
+
+	newExpectedSeq := getExpectedSeq(ipv4Packet, transportPacket)
+	length = uint32Distance(newExpectedSeq, expectedSeq)
+
+	if f.Network.A == ipv4Packet.SrcIP.String() {
+		f.Metric.ABPackets += int64(1)
+		f.Metric.ABBytes += int64(length)
+		f.state.aBExpectedSeq = newExpectedSeq
+	} else {
+		f.Metric.BAPackets += int64(1)
+		f.Metric.BABytes += int64(length)
+		f.state.bAExpectedSeq = newExpectedSeq
+	}
+
+	if transportPacket.FIN {
+		logging.GetLogger().Notice("FIN", f.state.aBExpectedSeq, f.state.bAExpectedSeq, f.Metric.ABBytes, f.Metric.BABytes)
```

Comment:
Notice is too verbose, you will see 1 log per tcp flow
As you don't dump the flowid I think you can remove this line